Eastern-Research-Group/mywaterway
# Feature/152 date slider

## Related Issues:
* [HMW-152](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-152)

## Main Changes:
* Created a Web Worker that parses data from the _Period of Record_ service, via **Papa Parse**.
* Added a date slider to the _Past Water Conditions_ sub-tab of the **Monitoring** tab on the **Community** page.
* Centralized initial updates to the `monitoringLocationsLayer` and the `monitoringGroups` context object to cut out duplicated code.
* Moved the `monitoringLocationsContent` JSX function in the `WaterbodyInfo` component into its own component, to better internalize and manage its own state.
* Added watchers for map mouse events, to propagate date slider changes to monitoring location map popups.

## Steps To Test:
1. Head over to the **Montoring** tab of the **Community** page
2. Click the _Past Water Conditions_ sub-tab
3. Once the slider is enabled (which should not take long), adjust the range and ensure that (a) toggle table values are being updated, (b) accordion items are being filtered and updated, (c) map graphics are being filtered, & (d) date ranges are being added to the download links under the toggle table, in each accordion item, and in each monitoring location pop-up (including selected locations in clusters.

**NOTE** The date ranges update upon release of the slider handle. Live updates are possible for HUCs with a relatively small amount of locations, but massively impacts performance in heavily monitored regions.
